merlin/schema/io/tensorflow_metadata.py

```diff
@@ -181,6 +181,12 @@ def from_merlin_schema(cls, schema: MerlinSchema) -> "TensorflowMetadata":
         """
         features = []
         for col_name, col_schema in schema.column_schemas.items():
+            if col_schema.dtype is None:
+                raise ValueError(
+                    f"Schema for column `{col_name}` does not have a dtype. "
+                    "ColumnSchema objects must have dtypes in order to be serializable."
+                )
+
```

Comment:
We do this validation here instead of the `ColumnSchema` constructor because it's sometimes convenient to allow column schemas to have blank dtypes (e.g. in the tests), but we can't serialize those column schemas to the Merlin schema format.